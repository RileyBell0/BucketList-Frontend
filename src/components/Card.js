import "../styles/card.css";
import React from "react";

function Card({ children, className, onClick }) {
  return (
    <div
      className={"card" + (className ? " " + className : "")}
      onClick={onClick}
    >
      {children}
    </div>
  );
}

export default Card;
