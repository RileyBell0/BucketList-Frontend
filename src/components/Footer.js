import { Link } from "react-router-dom";
import { useAuth } from "../auth/auth";
import "../styles/footer.css";

function Footer({ children, textClass, footerClass }) {
  const { handleLogout, authenticated } = useAuth();

  const text = "footer__text " + (textClass ? textClass : "");

  return (
    <>
      <div className="footer__placement">
        <div className="footer__children-container">{children}</div>
        <div
          className={"footer__container " + (footerClass ? footerClass : "")}
        >
          <div className="footer__content">
            <Link to="/" className="footer__link">
              <p className={text}>Welcome</p>
            </Link>
            <p className={text}> • </p>
            <Link to="/about" className="footer__link">
              <p className={text}>About Us</p>
            </Link>
            {authenticated && (
              <>
                <p className={text}> • </p>
                <Link to="/destinations" className="footer__link">
                  <p className={text}>Destinations</p>
                </Link>
                <p className={text}> • </p>
                <Link to="/trips" className="footer__link">
                  <p className={text}>Trips</p>
                </Link>
                <p className={text}> • </p>
                <Link to="/map" className="footer__link">
                  <p className={text}>Map</p>
                </Link>
                <p className={text}> • </p>
                <Link to="/help" className="footer__link">
                  <p className={text}>Help</p>
                </Link>
                <p className={text}> • </p>
                <Link to="/settings" className="footer__link">
                  <p className={text}>Settings</p>
                </Link>
                <p className={text}> • </p>

                <Link to="/" className="footer__link" onClick={handleLogout}>
                  <p className={text}>Sign Out</p>
                </Link>
              </>
            )}
            {!authenticated && (
              <>
                <p className={text}> • </p>
                <Link to="/login" className="footer__link">
                  <p className={text}>Log In</p>
                </Link>
                <p className={text}> • </p>
                <Link to="/signup" className="footer__link">
                  <p className={text}>Sign Up</p>
                </Link>
              </>
            )}
          </div>
        </div>
      </div>
    </>
  );
}

export default Footer;
