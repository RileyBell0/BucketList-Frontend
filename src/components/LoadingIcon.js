import "../styles/loadingIcon.css";
import { ReactComponent as Loading } from "../images/loading.svg";

function LoadingIcon({ className }) {
  return (
    <>
      <Loading className={"loading-icon " + className} />
    </>
  );
}

export default LoadingIcon;
