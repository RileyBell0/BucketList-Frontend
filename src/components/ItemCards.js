import React from "react";
import { ReactComponent as Delete } from "../images/delete.svg";
import { ReactComponent as Pin } from "../images/location_pin.svg";
import "../styles/ItemCards.css";
import { createSearchParams, useNavigate } from "react-router-dom";
import Card from "../components/Card";
import MediaQuery from "react-responsive";
import Globals from "../Globals";

function DestinationImageDesktop({ imgLink }) {
  return (
    <>
      {imgLink ? (
        //image from backend
        <div className="item-card__image--container">
          <img
            className="item-card__image"
            src={imgLink}
            alt="destination"
          ></img>
        </div>
      ) : null}
    </>
  );
}

function DestinationImageMobile({ imgLink }) {
  return (
    <>
      {imgLink ? (
        //image from backend
        <div className="item-card--mobile__image--container">
          <img
            className="item-card__image"
            src={imgLink}
            alt="destination"
          ></img>
        </div>
      ) : null}
    </>
  );
}

function ItemCard({ item, onDeletionIconClick, from }) {
  const navigate = useNavigate();

  let destSearchParams = {};
  if (from) {
    destSearchParams.from = from;
  }

  const generateDesktop = () => {
    return (
      <>
        <Card
          className="item-card"
          onClick={() => {
            navigate(
              {
                pathname: "/destinations/" + item._id,
                search: `?${createSearchParams(destSearchParams)}`,
              },
              { state: { dest: item } }
            );
          }}
        >
          <DestinationImageDesktop imgLink={item.imgLink} />
          <div className="item-card__content">
            <div className="item-card__content__header">
              <div className="item-card__content__header__text">
                <div className="item-card__content__header__text--name">
                  {item.name}
                </div>
                <div className="item-card__content__header__text--location">
                  {item.type}
                  {
                    // If a city/country doesn't exist, the string is just whitespace
                    String(item.city).trim() !== "" ||
                    String(item.country).trim() !== ""
                      ? " | "
                      : ""
                  }
                  {item.city}
                  {
                    // If a city/country doesn't exist, the string is just whitespace
                    String(item.city).trim() !== "" &&
                    String(item.country).trim() !== ""
                      ? ", "
                      : ""
                  }
                  {item.country}
                </div>
              </div>
              <div className="item-card__content__header--icons">
                <button
                  className="item-card__icons__icon"
                  onClick={(e) => {
                    e.stopPropagation();
                    onDeletionIconClick(item);
                  }}
                >
                  <Delete className="item-card__icons__delete" />
                </button>
              </div>
            </div>
            <div className="item-card__content__desc">
              {(item.desc !== "" && item.desc.replace("\n", "|")) ||
                (item.desc === "" && "No description was provided")}
            </div>
          </div>
        </Card>
      </>
    );
  };

  const generateMobile = () => {
    return (
      <>
        <Card
          className="item-card--mobile"
          onClick={() => {
            navigate(
              {
                pathname: "/destinations/" + item._id,
                search: `?${createSearchParams(destSearchParams)}`,
              },
              { state: { dest: item } }
            );
          }}
        >
          <div className="item-card--mobile__header">
            <DestinationImageMobile imgLink={item.imgLink} />
            <div className="item-card--mobile__header__content">
              <div className="item-card--mobile__header__text">
                <div className="item-card--mobile__header__text__name">
                  {item.name}
                </div>

                <div className="item-card--mobile__header__text__location">
                  {item.type}
                  {
                    // If a city/country doesn't exist, the string is just whitespace
                    String(item.city).trim() !== "" ||
                    String(item.country).trim() !== ""
                      ? " | "
                      : ""
                  }
                  {item.city}
                  {
                    // If a city/country doesn't exist, the string is just whitespace
                    String(item.city).trim() !== "" &&
                    String(item.country).trim() !== ""
                      ? ", "
                      : ""
                  }
                  {item.country}
                </div>
              </div>
              <div className="item-card--mobile__desc">
                {(item.desc !== "" && item.desc) ||
                  (item.desc === "" && "No description was provided")}
              </div>
            </div>
          </div>

          <div className="item-card--mobile__icons">
            <div
              className="item-card__icons__icon--container item-card__icons__icon--container--mobile"
              onClick={(e) => {
                e.stopPropagation();
                navigate({
                  pathname: "/map",
                  search: `?${createSearchParams({
                    lat: item.position[0].lat,
                    lng: item.position[0].lng,
                    zoom: 13,
                  })}`,
                });
              }}
            >
              <button className="item-card__icons__icon--mobile">
                <Pin className="item-card__icons__icon--map" />
              </button>
              <p className="item-card__icons__icon--text">View on Map</p>
            </div>
            <div
              className="item-card__icons__icon--container item-card__icons__icon--container--mobile"
              onClick={(e) => {
                e.stopPropagation();
                onDeletionIconClick(item);
              }}
            >
              <button className="item-card__icons__icon--mobile">
                <Delete className="item-card__icons__icon--delete" />
              </button>
              <p className="item-card__icons__icon--text">Delete</p>
            </div>
          </div>
        </Card>
      </>
    );
  };

  return (
    <>
      <MediaQuery minWidth={Globals.responsiveWidth + 1}>
        {generateDesktop()}
      </MediaQuery>
      <MediaQuery maxWidth={Globals.responsiveWidth}>
        {generateMobile()}
      </MediaQuery>
    </>
  );
}

function ItemCardList({ items, onDeletionIconClick, from }) {
  return (
    <div className="item-card-list">
      {items.map((item) => (
        <ItemCard
          key={item._id}
          item={item}
          onDeletionIconClick={onDeletionIconClick}
          from={from}
        />
      ))}
    </div>
  );
}

export { ItemCardList };
