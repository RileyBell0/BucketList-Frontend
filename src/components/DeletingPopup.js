import "../styles/deletingPopup.css";
import LoadingIcon from "./LoadingIcon";
import Popup from "./Popup";
import Card from "./Card";

function DeletingPopup() {
  return (
    <Popup>
      <div className="deleting-popup">
        <Card className="deleting-popup--container">
          <h1 className="deleting-popup--text">Deleting...</h1>
          <LoadingIcon />
        </Card>
      </div>
    </Popup>
  );
}

export default DeletingPopup;
