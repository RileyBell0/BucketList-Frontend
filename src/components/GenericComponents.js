import React from "react";
import "../styles/GenericComponents.css";
import { ReactComponent as Placeholder } from "../images/picture-icon.svg";
import { useState } from "react";
import Card from "../components/Card";
import { ReactComponent as ToggleOn } from "../images/toggle-on.svg";
import { ReactComponent as ToggleOff } from "../images/toggle-off.svg";
import { useDropzone } from "react-dropzone";
import TextAreaAutosize from "react-textarea-autosize";
import LoadingIcon from "../components/LoadingIcon";

function GenericInput(props) {
  return (
    <div className="generic-input">
      {/* Only draw a title if provided with one */}
      {props.title != null && (
        <p className="generic-field__name">{props.title}</p>
      )}

      <input
        className={
          "generic-input__input" +
          (props.error ? " generic-input__input--error" : "")
        }
        type={props.type}
        defaultValue={props.defaultValue}
        name={props.name}
        onBlur={props.onBlur}
        onChange={props.onChange}
        placeholder={props.placeholder}
        value={props.value}
        ref={props.reference}
      />
      {props.error && <div className="generic-input__error">{props.error}</div>}
    </div>
  );
}

function GenericDropdown({ title, options, onChange, defaultOption, value }) {
  const options_rendered = options.map((option) => (
    <option key={option}>{option}</option>
  ));

  return (
    <>
      <div>
        {/* Only draw a title if provided with one */}
        {title != null && <p className="generic-field__name">{title}</p>}
        <div className="generic-drop-down__container">
          <select
            className="generic-drop-down"
            defaultValue={options[defaultOption]}
            onChange={onChange}
            value={value}
          >
            {options_rendered}
          </select>
        </div>
      </div>
    </>
  );
}

function GenericButton(props) {
  return (
    <div>
      <button
        className={"generic-button " + props.className}
        onClick={props.onClick}
        type={props.type}
        style={{ maxWidth: props.maxWidth }}
        disabled={props.disabled}
      >
        {props.children}
      </button>
    </div>
  );
}

function GenericInputParagraph(props) {
  return (
    <>
      {/* Only draw a title if provided with one */}
      {props.title != null && (
        <p className="generic-field__name">{props.title}</p>
      )}

      <TextAreaAutosize
        className={
          "generic-input-paragraph" +
          (props.error ? " generic-input__input--error" : "")
        }
        defaultValue={props.defaultValue}
        onChange={props.onChange}
        type={props.type}
        name={props.name}
        onBlur={props.onBlur}
        placeholder={props.placeholder}
        value={props.value}
        ref={props.reference}
      />
      {props.error && <div className="generic-input__error">{props.error}</div>}
    </>
  );
}

function GenericParagraph({
  children,
  title,
  className,
  defaultValue,
  onClick,
}) {
  return (
    <>
      {/* Only draw a title if provided with one */}
      {title != null && <p className="generic-field__name">{title}</p>}
      <p
        className={
          "generic-paragraph__container generic-paragraph " + className
        }
        defaultValue={defaultValue}
        onClick={onClick ? onClick : () => {}}
      >
        {(children !== "" && children) ||
          /* Drawing a default value if no value was but a default value was */
          (children === "" && defaultValue !== undefined && defaultValue) ||
          (children === "" && defaultValue === undefined && " ")}
      </p>
    </>
  );
}

function GenericImageInput({
  image,
  prevImage,
  title,
  onChange,
  seamless,
  loading,
}) {
  const [showModal, setShowModal] = useState(false);
  const [errorType, setErrorType] = useState("");
  const [errorDetail, setErrorDetail] = useState("");
  const { getRootProps, getInputProps } = useDropzone({
    multiple: false,
    onDrop: (acceptedFile, fileRejections) => {
      if (fileRejections.length !== 0) {
        fileRejections.forEach((file) => {
          file.errors.forEach((error) => {
            setErrorDetail("Error: " + error.message);
            setShowModal(true);
          });
        });
      } else {
        CheckImage(acceptedFile, true);
      }
    },
  });

  const CheckImage = (inputFile, fromDragAndDrop) => {
    let file = inputFile[0];
    if (!fromDragAndDrop) {
      file = inputFile.target.files[0];
    }
    onChange(null);

    if (file.size > 5 * 1024 * 1024) {
      // e.target.value = "";
      setErrorType("Image is Too Big");
      setErrorDetail("Maximum file size is 5MB");
      setShowModal(true);
      return;
    }

    if (!file.type.match("image.*")) {
      // e.target.value = "";
      setErrorType("File is Invalid");
      setErrorDetail("Must be an Image");
      setShowModal(true);
      return;
    }

    if (file.size <= 5 * 1024 * 1024) {
      onChange(file);
    }
  };

  const RemoveImage = (e) => {
    e.preventDefault();
    e.stopPropagation();
    onChange(null);
  };

  const ImageModal = () => {
    return (
      <div className="transparent-layer">
        <div className="image-modal">
          <Card>
            <p>{errorType}</p>
            <p>{errorDetail}</p>
            <br />
            <GenericButton onClick={() => setShowModal(false)}>
              Okay
            </GenericButton>
          </Card>
        </div>
      </div>
    );
  };

  let imgClass = "generic-image-input__image-container__image";
  let overallClass = "generic-image-input";
  let containerClass = "generic-image-input__image-container";
  let removeImgIconClass = "generic-image-input__image-container__removeImage";
  let addImgIconClass = "generic-image-input__image-container__add-image";

  if (seamless === true) {
    imgClass =
      imgClass + " generic-image-input__image-container__image--seamless";
    overallClass = overallClass + " generic-image-input--seamless";
    containerClass =
      containerClass + " generic-image-input__image-container--seamless";
  }

  if (loading) {
    imgClass = imgClass + " generic-image-input__image--loading";
    overallClass = overallClass + " generic-image-input--loading";
  }

  return (
    <>
      {showModal ? <ImageModal /> : null}

      {/* Only draw a title if provided with one */}
      {title != null && <p className="generic-field__name">{title}</p>}

      {loading !== true && (
        <div
          {...getRootProps({
            className: overallClass,
          })}
        >
          <input {...getInputProps()} />
          {seamless !== true && <p>Click to Upload or Drag and Drop</p>}
          <div className={containerClass}>
            {image ? (
              //image that the user submits
              <>
                <button
                  className={removeImgIconClass}
                  type="button"
                  onClick={RemoveImage}
                >
                  X
                </button>
                <img
                  className={imgClass}
                  src={URL.createObjectURL(image)}
                  alt="user submitted"
                ></img>
              </>
            ) : prevImage ? (
              //imgur link if there is already a image submitted to the destination
              <>
                <button
                  className={removeImgIconClass}
                  type="button"
                  onClick={RemoveImage}
                >
                  X
                </button>
                <img
                  className={imgClass}
                  src={prevImage}
                  alt="placeholder"
                ></img>
              </>
            ) : (
              //placeholder image when user hasnt submitted an image and there wasn't an imgur link
              <>
                <button className={addImgIconClass} type="button">
                  +
                </button>
                <Placeholder className={imgClass}></Placeholder>
              </>
            )}
          </div>
          {seamless !== true && (
            <>
              <p className="generic-image-input__max-size">5 MB limit</p>
              <br />
            </>
          )}
        </div>
      )}
      {loading && (
        <>
          <div className={overallClass}>
            {seamless !== true && <p>Uploading Image...</p>}

            <div className={containerClass}>
              {image ? (
                //image that the user submits
                <>
                  <LoadingIcon className="generic-image-input__loading-icon" />
                  <img
                    className={imgClass}
                    src={URL.createObjectURL(image)}
                    alt="user submitted"
                  ></img>
                </>
              ) : prevImage ? (
                //imgur link if there is already a image submitted to the destination
                <>
                  <LoadingIcon className="generic-image-input__loading-icon" />
                  <img
                    className={imgClass}
                    src={prevImage}
                    alt="placeholder"
                  ></img>
                </>
              ) : (
                //placeholder image when user hasnt submitted an image and there wasn't an imgur link
                <>
                  <LoadingIcon className="generic-image-input__loading-icon" />
                  <Placeholder className={imgClass}></Placeholder>
                </>
              )}
            </div>
            {seamless !== true && (
              <p className="generic-image-input__max-size">Please Wait...</p>
            )}
          </div>
        </>
      )}
    </>
  );
}

function GenericToggle({ state, onClick, className, title, disabled }) {
  if (className === undefined) {
    className = "";
  }

  return (
    <>
      {title != null && <p className="generic-field__name">{title}</p>}
      <div className="generic-toggle">
        <div className="generic-toggle__button-container">
          {state && (
            <ToggleOn
              className={
                disabled
                  ? "generic-toggle__button generic-toggle__button-disabled " +
                    className
                  : "generic-toggle__button generic-toggle__button-active " +
                    className
              }
              onClick={() => {
                if (disabled !== true) {
                  onClick();
                }
              }}
            />
          )}
          {!state && (
            <ToggleOff
              className={
                disabled
                  ? "generic-toggle__button generic-toggle__button-disabled " +
                    className
                  : "generic-toggle__button generic-toggle__button-active " +
                    className
              }
              onClick={() => {
                if (disabled !== true) {
                  onClick();
                }
              }}
            />
          )}
        </div>
      </div>
    </>
  );
}

export {
  GenericInput,
  GenericDropdown,
  GenericButton,
  GenericParagraph,
  GenericInputParagraph,
  GenericImageInput,
  GenericToggle,
};
