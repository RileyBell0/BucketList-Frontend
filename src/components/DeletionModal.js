import { GenericButton } from "./GenericComponents";
import Card from "./Card";
import "../styles/deletionModal.css";
import Popup from "./Popup";

function DeletionModal({
  itemName,
  onDeletionModalConfirmation,
  onDeletionModalCancel,
}) {
  return (
    <>
      <Popup disablePopup={onDeletionModalCancel} z_index={998}>
        <div className="deletion-modal__content">
          <Card className="deletion-modal">
            <div>
              <p className="deletion-modal__text">
                Are you sure you want to delete
              </p>
              <p className="deletion-modal__text--highlight"> {itemName}</p>
              <p className="deletion-modal__text">?</p>
            </div>
            <br />
            <div className="deletion-modal__buttons">
              <GenericButton onClick={onDeletionModalCancel}>
                Cancel
              </GenericButton>
              <GenericButton
                onClick={onDeletionModalConfirmation}
                className="generic-button--warning"
              >
                Delete
              </GenericButton>
            </div>
          </Card>
        </div>
      </Popup>
    </>
  );
}

export { DeletionModal };
