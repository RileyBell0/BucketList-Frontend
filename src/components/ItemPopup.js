import "../styles/itemPopup.css";
import "../styles/ItemCards.css";
import { useNavigate } from "react-router-dom";
import Card from "../components/Card";
import { ReactComponent as Delete } from "../images/delete.svg";
import { ReactComponent as Edit } from "../images/edit.svg";
import { GenericButton, GenericParagraph } from "./GenericComponents";
import Popup from "./Popup";

function DestinationImage({ imgLink }) {
  return (
    <>
      {/* Draw the location's image next */}
      {imgLink ? (
        <>
          <br />
          <div className="item-popup__content_image-container">
            <img
              src={imgLink}
              alt="destination"
              className="item-popup__content__image"
            />
          </div>
        </>
      ) : (
        //image when there is no image in the backend
        <></>
      )}
    </>
  );
}

function ItemPopup({ item, disablePopup, onDeletionIconClick }) {
  const navigate = useNavigate();

  return (
    <>
      <Popup disablePopup={disablePopup}>
        <div className="item-popup__content">
          <Card className="item-popup__content__card">
            {/* Headings */}
            <h1 className="item-popup__content_name">{item.name}</h1>
            <h2>
              {item.city}
              {
                // If a city/country doesn't exist, the string is just whitespace
                String(item.city).replace(" ", "") !== "" &&
                String(item.country).replace(" ", "") !== ""
                  ? ", "
                  : ""
              }
              {item.country}
            </h2>
            <DestinationImage imgLink={item.imgLink} />

            {/* Description */}
            <h3>
              {item.type} - {item.trip ? item.trip : "None"}
            </h3>
            <GenericParagraph
              className="item-popup__content__description"
              defaultValue={"No description was provided"}
            >
              {item.desc}
            </GenericParagraph>

            {/* Buttons and Close */}
            <br />
            {/* Draw the buttons for editing this destination */}
            <div className="item-popup__icons">
              {/* Edit */}
              <div
                className="item-popup__icons__icon--container"
                onClick={() => {
                  navigate("/destinations/" + item._id);
                }}
              >
                <button className="item-popup__icons__icon">
                  <Edit className="item-popup__icons__icon--edit" />
                </button>
                <p className="item-popup__icons__icon--text">
                  Edit Destination
                </p>
              </div>

              {/* Delete */}
              <div
                className="item-popup__icons__icon--container"
                onClick={() => onDeletionIconClick(item)}
              >
                <button className="item-popup__icons__icon">
                  <Delete className="item-popup__icons__icon--delete" />
                </button>
                <p className="item-popup__icons__icon--text">Delete</p>
              </div>
            </div>

            {/* Close window */}
            <br />
            <GenericButton onClick={disablePopup}>Close</GenericButton>
          </Card>
        </div>
      </Popup>
    </>
  );
}

export { ItemPopup };
