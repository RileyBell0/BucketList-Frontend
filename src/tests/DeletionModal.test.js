import renderer from "react-test-renderer";
import { DeletionModal } from "../components/DeletionModal";

it("Matches previous snapshot", () => {
  const comp = renderer.create(
    <DeletionModal
      itemName={"Destination"}
      onDeletionModalCancel={"None"}
      onDeletionModalConfirmation={"None"}
    />
  );

  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});
