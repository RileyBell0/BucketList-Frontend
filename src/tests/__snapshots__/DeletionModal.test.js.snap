// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches previous snapshot 1`] = `
Array [
  <div
    className="popup__background"
    onClick="None"
    style={
      Object {
        "zIndex": 998,
      }
    }
  />,
  <div
    className="popup__container "
    style={
      Object {
        "top": 0,
        "zIndex": 999,
      }
    }
  >
    <div
      className="deletion-modal__content"
    >
      <div
        className="card deletion-modal"
      >
        <div>
          <p
            className="deletion-modal__text"
          >
            Are you sure you want to delete
          </p>
          <p
            className="deletion-modal__text--highlight"
          >
             
            Destination
          </p>
          <p
            className="deletion-modal__text"
          >
            ?
          </p>
        </div>
        <br />
        <div
          className="deletion-modal__buttons"
        >
          <div>
            <button
              className="generic-button undefined"
              onClick="None"
              style={
                Object {
                  "maxWidth": undefined,
                }
              }
            >
              Cancel
            </button>
          </div>
          <div>
            <button
              className="generic-button generic-button--warning"
              onClick="None"
              style={
                Object {
                  "maxWidth": undefined,
                }
              }
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
